src/index.js

```diff
@@ -416,18 +419,49 @@ export default class AuthnWidget {
   async handleMfaDeviceChange(evt) {
     evt.preventDefault();
     let state = await this.store.getState();
+    if (this.pollPushNoticationState) {
+      clearTimeout(this.pollPushNoticationState);
+    }
     state.status = 'DEVICE_SELECTION_REQUIRED';
     this.render(this.store.getPreviousStore(), state);
   }
 
+  registerMfaUsePasscodeEventHandler() {
+    if (document.getElementById('usePasscode')) {
+      document.getElementById('usePasscode')
+        .addEventListener('click', this.handleMfaUsePasscode);
+    }
+  }
+
+  async handleMfaUsePasscode(evt) {
+    evt.preventDefault();
+    clearTimeout(this.pollPushNoticationState);
+    let storeState = this.store.getStore();
+    storeState.status = 'OTP_REQUIRED';
+    this.render(this.store.getPreviousStore(), storeState);
```

Comment:
When in this state, is the behavior okay if the user refreshes their browser? I would think they would go back to the old state and it would be okay. Can you confirm?

Reply:
Yes, when you refresh after clicking "Use Passcode" the widget re-syncs with the server and it goes back to PUSH_CONFIRMATION_WAITING. Which matches the P1 implemenation.

Reply:
awesome, thanks for confirming